syscalls.c

```diff
@@ -802,8 +801,9 @@ int syscalls_signalPost(void *ustack)
 	err = threads_sigpost(proc, t, signal);
 
 	proc_put(proc);
-	threads_put(t);
-	hal_cpuReschedule(NULL, NULL);
+	if (t != NULL)
+		threads_put(t);
+
```

Comment:
`if` I get, but why remove reschedule?

Reply:
Reschedule is already called inside `threads_sigpost()` - I think there is no need to call it again.

Reply:
Ok, makes sense